src/NuGetForUnity/Editor/NugetAssetPostprocessor.cs

```diff
@@ -191,6 +221,44 @@ private static void ModifyImportSettingsOfRoslynAnalyzer([NotNull] PluginImporte
 
             AssetDatabase.SetLabels(plugin, new[] { RoslynAnalyzerLabel, ProcessedLabel });
 
+            // The nuget package can contain analyzers for multiple Roslyn versions.
+            // In that case, for the same package, the most recent version must be chosen out of those available for the current Unity version.
+            var assetRoslynVersion = GetRoslynVersionNumberFromAnalyzerPath(plugin.assetPath);
+            if (assetRoslynVersion != null)
+            {
+                var versionPrefixIndex = plugin.assetPath.IndexOf(AnalyzersRoslynVersionsFolderName, StringComparison.Ordinal);
+                var analyzersVersionsRoot = JoinPathComponents(plugin.assetPath.Substring(0, versionPrefixIndex), AnalyzersRoslynVersionsFolderName);
+                var analyzersFolders = AssetDatabase.GetSubFolders(analyzersVersionsRoot);
+
+                var enabledRoslynVersions = analyzersFolders
+                    .Select(GetRoslynVersionNumberFromAnalyzerPath)
+                    .Where(ver =>
+                    {
+                        if (ver == null)
+                        {
+                            return false;
+                        }
+#if UNITY_2022_3_OR_NEWER && !UNITY_2022_3_1 && !UNITY_2022_3_2 && !UNITY_2022_3_3 && !UNITY_2022_3_3 && !UNITY_2022_3_4 && !UNITY_2022_3_5 && !UNITY_2022_3_6 && !UNITY_2022_3_7 && !UNITY_2022_3_8 && !UNITY_2022_3_9 && !UNITY_2022_3_10 && !UNITY_2022_3_11
+                        return string.CompareOrdinal(ver, "4.3.0") <= 0;
+#elif UNITY_2022_2_OR_NEWER
+                        return string.CompareOrdinal(ver, "4.1.0") <= 0;
+#elif UNITY_2021_2_OR_NEWER
+                        return string.CompareOrdinal(ver, "3.8.0") <= 0;
+#else
+                        return false;
+#endif
+                    })
+                    .ToArray();
+
+                // If most recent valid analyzers exist elsewhere, remove label `RoslynAnalyzer`
+                if (!enabledRoslynVersions.Contains(assetRoslynVersion) ||
+                    string.CompareOrdinal(assetRoslynVersion, enabledRoslynVersions.Max()) < 0)
+                {
+                    AssetDatabase.SetLabels(plugin, new[] { ProcessedLabel });
```

Comment:
Instead of setting and then removing the label, just save in some bool variable if RoslynAnalyzer label should be added and then in the end just SetLabels once to proper value depending on that bool.

Reply:
Ok, I fixed.